### PR TITLE
Add Instructional Resources to Reporting UI for Groups and School/Grade assessment displays

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
### Overall view of a group
screen shot 2017-12-12 at 6 06 29 pm

### Showing links 
screen shot 2017-12-12 at 6 11 29 pm

### Student score distribution (no links)
screen shot 2017-12-12 at 6 11 17 pm

### Datatable (no links)
screen shot 2017-12-12 at 6 14 12 pm

###  Changes to student history
screen shot 2017-12-12 at 6 11 46 pm

### CODE REVIEW

File: admin-webapp/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-/src/main/resources/static/
\ No newline at end of file
+/src/main/resources/static/
+/bin/
```
Comment (esotrope) [inline admin-webapp/.gitignore]: something get generated there?

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/instructionalresource/DefaultInstructionalResourceService.java [hunk 2]
```diff
@@ -20,14 +24,26 @@
...
+                repository.findByAssessmentAndSchool(assessmentId, schoolId).stream()
+                        .map(instructionalResource -> setStateName(instructionalResource))
+                        .collect(Collectors.toList())
+        );
+    }
+
+    private InstructionalResource setStateName(final InstructionalResource resource) {
+        return State.equals(resource.getOrganizationLevel())
+                ? InstructionalResource.builder().copy(resource).organizationName(applicationProperties.getState().getName()).build()
+                : resource;
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/instructionalresource/DefaultInstructionalResourceService.java]: looks like there is some code reuse ops with the state logic we can explore later maybe

File: webapp/src/main/webapp/src/app/assessments/model/instructional-resources.model.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export class InstructionalResources {
+  performanceLevelToResources: Map<number, InstructionalResource[]>;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/model/instructional-resources.model.ts]: nit: to be consistent with other places I would call this resourcesByPerformanceLevel

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/instructionalresource/DefaultInstructionalResourceService.java [hunk 4]
```diff
@@ -20,14 +24,26 @@
...
-    public DefaultInstructionalResourceService(final InstructionalResourceRepository repository) {
+    public DefaultInstructionalResourceService(final InstructionalResourceRepository repository, final ApplicationProperties applicationProperties) {
         this.repository = repository;
+        this.applicationProperties = applicationProperties;
     }
 
     @Override
     public List<InstructionalResource> findByAssessmentAndSchool(final long assessmentId, final long schoolId) {
-        return IROrdering.sortedCopy(repository.findByAssessmentAndSchool(assessmentId, schoolId));
+        return IROrdering.sortedCopy(
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/instructionalresource/DefaultInstructionalResourceService.java]: nit: would rename to `InstructionalResourceOrdering` not a fan of abbr

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/configuration/ApplicationProperties.java [hunk 5]
```diff
@@ -167,4 +176,31 @@ public void setUserGuide(final String userGuide) {
...
+        public String getCode() {
+            return code;
+        }
+
+        public void setCode(final String code) {
+            this.code = code;
+        }
+
+        /**
+         * @return the state code
```
Comment (wtritch) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/configuration/ApplicationProperties.java]: nit: the state name

File: webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<div *ngIf="resources !== undefined && resources.length > 0; else noResources">
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.html]: You could slightly simplify this to "resources && resources.length > 0"

File: webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+})
+export class InstructionalResourcePopoverComponent {
+
+  constructor(private translateService: TranslateService) {}
+
+  @Input()
+  resources: InstructionalResource[];
+
+  @Input()
+  performanceLevel: number;
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.ts]: Is this used by anything?

File: webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import { InstructionalResource } from "../model/instructional-resources.model";
+import { TranslateService } from "@ngx-translate/core";
+
+@Component({
+  selector: 'instructional-resource-popover',
+  templateUrl: './instructional-resource-popover.component.html'
+})
+export class InstructionalResourcePopoverComponent {
+
+  constructor(private translateService: TranslateService) {}
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.ts]: Don't think we need to inject the translateService here

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html [hunk 9]
```diff
@@ -19,21 +19,20 @@
...
-              </span>
+            <button
+              class="btn btn-success btn-xs float-none-xs pull-right"
+              angulartics2On="click"
+              angularticsEvent="InstructionalResource"
+              angularticsCategory="AssessmentResults"
+              [angularticsProperties]="{label: assessmentExam.assessment.label}"
+              [disabled]="false"
+              [popover]="buttonPopover"
+              popoverTitle="{{ 'labels.groups.results.assessment.instruct-button' | translate:{ level:(assessmentExam.assessment.label | translate) } }}"
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html]: This feels a little funky... I think we can just replace `(assessmentExam.assessment.label | translate)` with `assessmentExam.assessment.label` We probably shouldn't be trying to translate the assessment label.

File: webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.html [hunk 10]
```diff
@@ -65,17 +65,29 @@
...
+        angulartics2On="click"
+        angularticsEvent="InstructionalResource"
+        angularticsCategory="AssessmentResults"
+        [disabled]="false"
+        [popover]="buttonPopover"
+        popoverTitle="{{ 'labels.groups.results.assessment.instruct-button' | translate:{ level:((examLevelEnum + (exam.level ? exam.level : 'missing')) | translate) } }}"
+        container="body"
+        (click)="loadInstructionalResources(exam)"
+        [outsideClick]="true"><i class="fa fa-book"></i></button>
+      <!--<span book-button title="{{ examLevelEnum + (exam.level ? exam.level : 'missing') | translate }}" [level]="exam.level ? exam.level.toString() : 'missing'" [assessment]="assessment"></span>-->
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.html]: Recommend removing this.

File: webapp/src/main/webapp/src/app/shared/button/information-button.component.ts [hunk 11]
```diff
@@ -1,20 +1,23 @@
...
   @Input()
   public title: string;
 
   @Input()
   public content: string;
 
+  @Input()
+  public icon: string;
+
+
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/shared/button/information-button.component.ts]: nit: extra space

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html [hunk 12]
```diff
@@ -33,20 +36,36 @@
   </p-column>
   <p-column field="exam.level" sortable="true">
     <ng-template pTemplate="header">
-      <span info-label title="{{'labels.groups.results.assessment.exams.cols.iab.performance' | translate}}"
-                  content="{{'labels.groups.results.assessment.exams.cols.iab.performance-info' | translate}}"></span>
+      <span info-button title="{{'labels.groups.results.assessment.exams.cols.iab.performance' | translate}}"
+            content="{{'labels.groups.results.assessment.exams.cols.iab.performance-info' | translate}}"></span>
     </ng-template>
-    <ng-template let-exam="rowData.exam" pTemplate="body">
+    <ng-template let-i="rowIndex" let-exam="rowData.exam" pTemplate="body">
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html]: I don't think you need the `let-i="rowIndex"` here?

File: admin-webapp/.gitignore (hunk 1 above)
Comment (dsnyder-fairway) [reply admin-webapp/.gitignore]: yeah, it looks to be a copy of the compiled classes and static/assets

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html (hunk 12 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html]: switched to passing `rowData`

File: webapp/src/main/webapp/src/app/shared/button/information-button.component.ts (hunk 11 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/shared/button/information-button.component.ts]: removed

File: webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.html (hunk 10 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.html]: removed

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html (hunk 9 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html]: good call. Removed

File: webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.ts (hunk 8 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.ts]: you are correct. removed

File: webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.ts (hunk 7 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.ts]: nope, removed

File: webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.html (hunk 6 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.html]: done

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/configuration/ApplicationProperties.java (hunk 5 above)
Comment (dsnyder-fairway) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/configuration/ApplicationProperties.java]: updated

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/instructionalresource/DefaultInstructionalResourceService.java (hunk 4 above)
Comment (dsnyder-fairway) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/instructionalresource/DefaultInstructionalResourceService.java]: renamed

File: webapp/src/main/webapp/src/app/assessments/model/instructional-resources.model.ts (hunk 3 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/model/instructional-resources.model.ts]: updated

File: webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.html (hunk 6 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.html]: you can super-simplify it by doing `resource && resources.length` ; P

**Pull request merged**